infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -129,4 +157,46 @@ private void checkLimits() {
             }
         }
     }
+
+    public static class PartitionAgent {
+        @Getter
+        private PartitionAttribute partitionAttribute;
+        private final Path logPartition;
+        private final FileSystemConfig config;
+        public PartitionAgent(FileSystemConfig config) {
+            this.config =  config;
+            logPartition = Paths.get(config.logDir.getRoot().toString(), config.logDir.subpath(0, 1).toString());
+            partitionAttribute = setPartitionAttribute();
+            //we need a scheduler
+        }
+
+        private PartitionAttribute setPartitionAttribute() {
+            try {
+                File logDirectoryFile = config.logDir.toFile();
+                FileStore fileStore = Files.getFileStore(config.logDir);
+                partitionAttribute = new PartitionAttribute(
+                        fileStore.isReadOnly() || !logDirectoryFile.canWrite(),
+                        fileStore.getUsableSpace(),
+                        fileStore.getTotalSpace()
+                );
+            } catch (IOException e) {
+                log.error("setPartitionStats: Error while getting PartitionStats", e);
```

Comment:
Is it safe to ignore the IOException here?

Reply:
Let's ask Chetan in the 4-th pr